modules/core/src/main/java/org/apache/synapse/aspects/flow/statistics/tracing/opentelemetry/management/OTLPTelemetryManager.java

```diff
@@ -0,0 +1,106 @@
+package org.apache.synapse.aspects.flow.statistics.tracing.opentelemetry.management;
+
+import io.opentelemetry.api.OpenTelemetry;
+import io.opentelemetry.api.common.Attributes;
+import io.opentelemetry.api.trace.Tracer;
+import io.opentelemetry.context.propagation.ContextPropagators;
+import io.opentelemetry.exporter.otlp.trace.OtlpGrpcSpanExporter;
+import io.opentelemetry.exporter.otlp.trace.OtlpGrpcSpanExporterBuilder;
+import io.opentelemetry.extension.trace.propagation.JaegerPropagator;
+import io.opentelemetry.sdk.OpenTelemetrySdk;
+import io.opentelemetry.sdk.resources.Resource;
+import io.opentelemetry.sdk.trace.SdkTracerProvider;
+import io.opentelemetry.sdk.trace.export.BatchSpanProcessor;
+import io.opentelemetry.semconv.resource.attributes.ResourceAttributes;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.apache.synapse.aspects.flow.statistics.tracing.opentelemetry.management.handling.span.OpenTelemetrySpanHandler;
+import org.apache.synapse.aspects.flow.statistics.tracing.opentelemetry.management.handling.span.SpanHandler;
+import org.apache.synapse.aspects.flow.statistics.tracing.opentelemetry.management.scoping.TracingScopeManager;
+import org.apache.synapse.config.SynapsePropertiesLoader;
+
+import java.util.Enumeration;
+
+public class OTLPTelemetryManager implements OpenTelemetryManager {
+
+    private Log logger = LogFactory.getLog(OTLPTelemetryManager.class);
+    private SdkTracerProvider sdkTracerProvider;
+    private OpenTelemetry openTelemetry;
+    private TelemetryTracer tracer;
+    private SpanHandler handler;
+
+    @Override
+    public void init() {
+
+        String headerProperty = getHeaderKeyProperty();
```

Comment:
`getHeaderKeyProperty()` can return `null`. Shall we handle that, if `headerProperty == null` we throw an error saying that the required property is not found?

Reply:
Yeah we can but it won't be null unless the properties are not set through toml file by mistakenly as the header and end point url is required.

Reply:
Resolved after discussion.